### PR TITLE
Feature Ranking

### PR DESCRIPTION
**Pull request by neelasha23**
Added 1D and 2D feature ranking

### CODE REVIEW

**Comment by neelasha23** [comment]
I have a few queries:

1. I have added a method `plot_feature_ranks_custom_algorithm` which wasn't present in the `yellowbricks` repo. But this method might help if user wants to user custom ranking algorithm and visualise the rankings generated. Is this method fine or I should remove?
2. I have added docstring to the `Rank1D` and `Rank2D` classes. Do we need to add docstring to the base class `RankD` as well?
3. The method `plot_feature_ranks_custom_algorithm` is not rendered on the docs as it's a part of the base class `RankD`. So should I just include corresponding usage of this method in the user guide?

**Comment by idomic** [comment]
1. I think it's fine.
2. Yes go ahead.
3. If this function is exposed to the user it should be documented. If it's an internal helper function we can skip it.

**Comment by idomic** [comment]
@yafimvo please review

File: examples/feature_ranking_2D.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import matplotlib.pyplot as plt
+from sklearn_evaluation.plot import Rank2D
+
+from sklearn.datasets import load_breast_cancer as load_data
+
+X, y = load_data(return_X_y=True)
+
+features = [
```
Comment (idomic) [inline examples/feature_ranking_2D.py:8]: Most of it is duplicated from 1D right?
Maybe it's worth combining the examples?

File: src/sklearn_evaluation/plot/feature_ranking.py [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+        """
+
+        self._validate_rank(ranks)
+        self._derive_features_from_ranks(ranks)
+        self.ranks_ = ranks
+        self._draw()
+        return self.ax
+
+
+class Rank1D(RankD):
```
Comment (idomic) [inline src/sklearn_evaluation/plot/feature_ranking.py:193]: The classes look very similar, does it make sense to use an abstract class/interface here?

**Review by idomic — CHANGES_REQUESTED** [review]
Added a few comments.
Also, looks like the doc guide is missing?

File: examples/feature_ranking_2D.py (hunk 1 above)
Comment (neelasha23) [reply examples/feature_ranking_2D.py:8]: The example scripts are rendered in the docs, like this:
```
Examples
    ---------
    .. plot:: ../../examples/feature_ranking_1D.py
```

So examples are separate for 1D and 2D

File: src/sklearn_evaluation/plot/feature_ranking.py (hunk 2 above)
Comment (neelasha23) [reply src/sklearn_evaluation/plot/feature_ranking.py:193]: `RankD` is the base class and has all the common functionalities. The only different implementation is `_draw` method, I can move it as abstract to the base class.

**Comment by neelasha23** [comment]
@idomic The guide is [here](https://github.com/ploomber/sklearn-evaluation/blob/98cdf344691ae588bdaf2ac799d5be1cab43970f/docs/source/user_guide/feature_ranking.ipynb)

**Comment by yafimvo** [comment]
Tested the code and docs locally - lgtm

File: src/sklearn_evaluation/plot/feature_ranking.py (hunk 2 above)
Comment (yafimvo) [reply src/sklearn_evaluation/plot/feature_ranking.py:193]: Would add `draw` to RankD for more possible implementations

**Pull request merged**